src/core/net/udp6.cpp

```diff
@@ -527,6 +527,18 @@ void Udp::HandlePayload(Message &aMessage, MessageInfo &aMessageInfo)
     return;
 }
 
+bool Udp::IsPortInUse(uint16_t aPort) const
+{
+    for (const SocketHandle * socket = mSockets.GetHead(); socket != nullptr; socket=socket->GetNext())
+    {
+        if(socket->GetSockName().GetPort() == aPort)
+        {
+            return true;
+        }
+    }
+    return false;
+}
```

Comment:
Here we are going through all the sockets in the list. The list here includes sockets added by OT core itself (e.g., internal Thread use), but I think it also contains sockets that may be added by user of OT APIs, from `otUdpOpen()`. So I think we would be then filtering all port used by any UDP socket (internal by core of external through `otUdpOpen()`).

Reply:
My intention is to filter any traffic initiated by OpenThread and all of its API. Unless I am missing some bigger picture here, when port is opened using otUdpOpen() the communication is handled using otUdpAddReceiver, otUdpSent, etc. and does not need to be passed to mReceiveIp6DatagramCallback.
In our case we noticed an issue with zephyr:
When OpenThread API is used to handle CoAP it opens an udp port. Any frame received is first handled by OpenThread, but after that it is passed to Zephyr and since the port is not opened there it responds with "ICMPv6 Destination Unreachable (Port unreachable)".